src/acl/Checklist.cc

```diff
@@ -218,7 +203,7 @@ ACLChecklist::changeState (AsyncState *newState)
      * relax this once conversion to states is complete
      * RBC 02 2003
      */
-    assert (state_ == NullState::Instance() || newState == NullState::Instance());
+    assert (!state_ || !newState);
     state_ = newState;
```

Comment:
Can you explain why we need to _store_ this state (i.e. the function pointer)? If we store it, it feels like we should use the stored value at some point (beyond assertions), but I failed to find that point. We use matchPath to find the ACL that we need to resume checking after we goAsync().

Needless to say, if we no longer need ACLChecklist::state_ member, then this PR should probably remove it because this PR changes it so much (even if that member became effectively unused earlier).

Reply:
Yes, this member is unused (except for a couple of assertions) - so I am removing it.